buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/task-properties-validation.gradle.kts

```diff
@@ -22,54 +22,21 @@ plugins {
 val validateTaskName = "validatePlugins"
 val reportFileName = "task-properties/report.txt"
 
-afterEvaluate {
-    // This is in an after evaluate block to defer the decision until after the `java-gradle-plugin` may have been applied, so as to not collide with it
-    // It would be better to use some convention plugins instead, that apply a fixes set of plugins (including this one)
-    if (plugins.hasPlugin("java-base")) {
-        val validationRuntime by configurations.creating {
-            isCanBeConsumed = false
-            attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.JAVA_RUNTIME))
-            // Required to select the right :distributions variant
-            attributes {
-                attribute(Attribute.of("org.gradle.runtime", String::class.java), "minimal")
-                attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.LIBRARY))
-                attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements.JAR))
-            }
-        }
-        dependencies {
-            // use a minimal gradle runtime here
-            validationRuntime(project(":core"))
-            validationRuntime(project(":dependencyManagement"))
-            validationRuntime(project(":platformJvm"))
-        }
-
-        val validateTask = if (plugins.hasPlugin("java-gradle-plugin")) {
-            tasks.named(validateTaskName, ValidatePlugins::class)
-        } else {
-            tasks.register(validateTaskName, ValidatePlugins::class)
-        }
-
-        validateTask {
-            configureValidateTask(validationRuntime)
-        }
-        tasks.named("codeQuality") {
-            dependsOn(validateTask)
-        }
-        tasks.withType(Test::class).configureEach {
-            shouldRunAfter(validateTask)
-        }
-    }
+val validateTask = tasks.register<ValidatePlugins>(validateTaskName)
+validateTask {
+    configureValidateTask()
+}
+tasks.named("codeQuality") {
+    dependsOn(validateTask)
+}
+tasks.withType<Test>().configureEach {
+    shouldRunAfter(validateTask)
 }
 
-fun ValidatePlugins.configureValidateTask(validationRuntime: Configuration) {
-    val main by project.sourceSets
-    dependsOn(main.output)
-    classes.setFrom(main.output.classesDirs)
-    classpath.from(main.output) // to pick up resources too
+fun ValidatePlugins.configureValidateTask() {
+    val main = project.sourceSets.main.get()
+    classes.from(main.output)
     classpath.from(main.runtimeClasspath)
-    classpath.from(validationRuntime)
```

Comment:
Why is the `validationRuntime` classpath not necessary anymore? The idea was to use the `PropertyValidationAccess.collectValidationProblems` code which is currently on `master`. I don't know if that will actually work now given https://github.com/gradle/gradle/issues/13678, but at least that was the idea.

Reply:
I am not sure why it was there in the first place. Everything we need for the classes under validation is provided by 'main.runtimeClasspath'.

If the idea was to run the worker action in the task from the current distribution it is not working like this. At least not anymore. That's why I removed it.

We somehow would need to force using the worker action class from the current distribution. I would suggest to follow up in #13678.

Reply:
Let's try to fix this as part of #13678.